diff --git a/launch/pal_gazebo.launch.py b/launch/pal_gazebo.launch.py
index a12238d..0e035ac 100644
--- a/launch/pal_gazebo.launch.py
+++ b/launch/pal_gazebo.launch.py
@@ -111,7 +111,7 @@ def start_gz(context, *args, **kwargs):
     world = find_world(world_name, priv_pkg_path, pkg_path, '.sdf')

     # Command to start the gazebo server.
-    gazebo_server_cmd_line = ['ign', 'gazebo', '-r', '-v', '4', '-s', world]
+    gazebo_server_cmd_line = ['gz', 'sim', '-r', '-v', '4', '-s', world]
     # Start the server under the gdb framework.
     debug = LaunchConfiguration('debug').perform(context)
     if debug == 'True':
@@ -123,7 +123,7 @@ def start_gz(context, *args, **kwargs):
         cmd=gazebo_server_cmd_line, output='screen')

     start_gazebo_client_cmd = ExecuteProcess(
-        cmd=['ign', 'gazebo', '-v', '4', '-g'], output='screen',
+        cmd=['gz', 'sim', '-v', '4', '-g'], output='screen',
         condition=IfCondition(LaunchConfiguration('gzclient'))
     )

